docs/platforms/javascript/common/enriching-events/transaction-name/index.mdx

```diff
@@ -22,15 +40,104 @@ task being executed. For example:
 Ideally, the transaction name does not contain variable values such as user
 IDs but has rather low cardinality while still uniquely identifying a piece of
 code you care about.
+For example, instead of `'GET /api/users/123/details'`, set `'GET /api/users/:id/details'`.
+This ensures that errors or traces for all requests made to this route are grouped together.
+
+### When to set the Transaction Name
+
+We generally recommend to let the SDK set the transaction name automatically.
+However, in some cases, the SDK might not be able to set a name or the set name doesn't work for your use case.
+
+If you need to set the transaction name manually, we recommend setting it _as early as possible_ in your application code, specifically when the operation you want to describe with the transaction name starts.
+This ensures that errors thrown early in the life cycle of your application are correctly annotated.
+Furthermore, it ensures consistency across a trace if you use [distributed tracing](/concepts/key-terms/tracing/distributed-tracing/).
+
+For example, in a server application, you might set the transaction name in a middleware that runs before the request is processed.
+In a web application, you might set the transaction name when you enter a new route (e.g. in an SPA Router).
+
```

Comment:
```suggestion

It's usually best to let the SDK set the transaction name automatically. However, if it can't or the name doesn't suit your needs, you can set it manually.

We recommend setting it _as early as possible_ in your application code, ideally when the operation you want to track starts. This ensures that errors thrown early in the application lifecycle are annotated correctly and keeps traces consistent if you're using [distributed tracing](/concepts/key-terms/tracing/distributed-tracing/).

For example, in a server app, you can set the transaction name in middleware before handling a request.
In a web app, you can set it when navigating to a new route, like in an SPA router.

```